content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,141 @@
+
+## Prerequisites
+
+To use Prisma Client extensions, you must enable the `clientExtensions` preview feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Multiple extensions
+
+Each time that you call `$extends`, you create an _extended client_. This extended client is a forked variant of the standard Prisma Client. The original client does not change.
+
+Every extended client operates independently, in an isolated instance. It cannot conflict with other extended clients or the original client.
+
+You can also combine two or more extensions into one extension, if you want the functionality from both extensions to apply in the same client instance. Note that it is possible for [combined extensions to conflict](#conflicts-in-combined-extensions).
+
+### Examples of isolated client instances
+
+Example uses of isolated client instances include the following:
+
+- Implement row-level security (RLS), where each http request has its own client with its own RLS extension, customized with session data. This can keep each user entirely separate, each in a separate client.
+- Add a `user.current()` method for the `user` model to get the current logged-in user
+- Enable more verbose logging for requests if a debug cookie is set
+- Attach a unique request id to all logs so you can correlate them later
+- Remove a `delete` method from models unless the application calls the admin endpoint and the user has the necessary priviliges
+
+### Declare multiple extensions
+
+In the following example, suppose that you have two extensions, `extensionA` and `extensionB`. There are two ways to combine these into a client.
+
+First, declare the standard Prisma client as usual:
+
+```ts
+// First of all, store your original Prisma Client in a variable as usual
+const prisma = new PrismaClient()
+```
+
+** Option 1: declare the new client in one line **
+
+With this option, you apply both extensions to the standard client in one line of code.
+
+```ts
+//Declare an extended client that has extensionsA and extensionB
+const clientAB = prisma.$extends(extensionA).$extends(extensionB)
+```
+
+You can refer to `clientAB` in your code, for example `clientAB.getName()`.
+
+** Option 2: declare multiple separate clients **
+
+The advantage of this option is that you can call any of the extended clients separately.
+
+```ts
+// Declare an extended client that has extension A applied
+const clientA = prisma.$extends(extensionA)
+
+// Declare an extended client that has extension B applied
+const clientB = prisma.$extends(extensionB)
+
+//Declare an extended client that is a combination of clientA and extensionB
```

Comment:
```suggestion
// Declare an extended client that has extensionsA and extensionB
const clientAB = prisma.$extends(extensionA).$extends(extensionB)
```

You can refer to `clientAB` in your code, for example `clientAB.getName()`.

** Option 2: declare multiple separate clients **

The advantage of this option is that you can call any of the extended clients separately.

```ts
// Declare an extended client that has extension A applied
const clientA = prisma.$extends(extensionA)

// Declare an extended client that has extension B applied
const clientB = prisma.$extends(extensionB)

// Declare an extended client that is a combination of clientA and extensionB
```

Reply:
@janpio  I can't work out what your suggestion is here :-( Can you take another look tomorrow morning, when I've finished pushing my changes this evening?

Reply:
GitHub has trouble when you make a suggestion for text that include a driple backticks :D Not it is fixed: Only fix a missing a space and a wrong variable name in comment.